File: go/space-age/space_age.go
//Package space converts age from seconds to years (based on the planet's orbital cycle)
package space

//A Planet is a string
type Planet string

//EarthYears returns the number of earth years given the seconds as an argument
func EarthYears(seconds float64) float64 {
	return seconds / 31557600
}

//Age returns the number of planet years
func Age(seconds float64, planet Planet) float64 {
	planetsToEarthYears := map[Planet]float64{
		"Earth":   1,
		"Mercury": 0.2408467,
		"Venus":   0.61519726,
		"Mars":    1.8808158,
		"Jupiter": 11.862615,
		"Saturn":  29.447498,
		"Uranus":  84.016846,
		"Neptune": 164.79132,
	}
	earthYears := EarthYears(seconds)
	planetYears := earthYears / planetsToEarthYears[planet]
	// fmt.Printf(
	// 	"%v seconds == %v Earth Years == %v %v years\n",
	// 	seconds,
	// 	earthYears,
	// 	planetYears,
	// 	planet,
	// )
	return planetYears
}

File: ruby/space-age/space_age.rb
class SpaceAge

  EARTH_YEARS_TO_PLANET_YEARS = {
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  }

  def initialize(time_in_seconds)
    @time_in_seconds = time_in_seconds.to_f
  end

  def on_earth
    @time_in_seconds / 31557600
  end

  EARTH_YEARS_TO_PLANET_YEARS.each do |planet, orbital_period|
    define_method("on_#{planet}") do
      on_earth / orbital_period
    end
  end

end
